Sources/TensorFlow/Optimizers/Optimizer.swift

```diff
@@ -12,12 +12,15 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+public typealias Optimizable =
+    VectorProtocol & ElementaryFunctions & PointwiseMultiplicative & KeyPathIterable
+
 /// A numerical optimizer.
 ///
 /// Optimizers apply an optimization algorithm to update the differentiable models.
 public protocol Optimizer {
     /// The type of the model whose parameters are optimized.
-    associatedtype Model: Differentiable
+    associatedtype Model: Differentiable where Model.TangentVector: Optimizable
```

Comment:
I feel such a requirement makes optimizers no longer general in it forces optimizers to conform to a series of auxiliary protocols, while each optimizer has quite different requirements. I think we should think about the proper Swift Evolution route: propose proper changes to the protocol hierarchy so that a conformance constraint to a certain protocol gives more available operations.

Reply:
Yes that's a good point. In this case, I can revert back to the specific requirements of each optimizer and only have layer tangent vectors conform to all of those for now, so that they can be used seamlessly with all optimizers. How does that sound as a temporary solution?